src/prng/prng.lisp

```diff
@@ -27,27 +20,42 @@
 
 (defmethod make-prng :around (name &key (seed :random))
   (let ((prng (call-next-method)))
-    (cond
-      ((eq seed nil))
-      ((find seed '(:random :urandom)) (read-os-random-seed seed prng))
-      ((or (pathnamep seed) (stringp seed)) (read-seed seed prng))
-      ((typep seed 'simple-octet-vector)
-       (reseed (slot-value prng 'generator) seed)
-       (incf (slot-value prng 'reseed-count)))
-      (t (error "SEED must be an octet vector, pathname indicator, :random or :urandom")))
+    (unless (find name '(:os :fortuna-generator))
+      (cond
+        ((eq seed nil))
+        ((find seed '(:random :urandom)) (read-os-random-seed seed prng))
+        ((or (pathnamep seed) (stringp seed)) (read-seed seed prng))
+        ((typep seed 'simple-octet-vector)
+         (prng-reseed seed prng)
+         (incf (slot-value prng 'reseed-count)))
+        (t (error "SEED must be an octet vector, pathname indicator, :random or :urandom"))))
```

Comment:
Making a Fortuna PRNG with a ```:random``` seed doesn't work anymore because of a failed assertion:

``` lisp
(ironclad:make-prng :fortuna :seed :urandom)

The assertion
(= #1=(LENGTH IRONCLAD::SEED) IRONCLAD::+FORTUNA-SEED-LENGTH+)
failed with #1# = 0, IRONCLAD::+FORTUNA-SEED-LENGTH+ = 64.
   [Condition of type SIMPLE-ERROR]
```

If allowing only a determined octet vector as seed is the expected behaviour (to create a deterministic pseudo-random sequence of bytes), it could be useful to change the documentation and error strings of the ```make-prng``` function and remove the references to ```:random``` and ```:urandom``` as they won't be used neither by Fortuna nor by OS-PRNG.